js/debug/job_recommendation.js

```diff
@@ -0,0 +1,53 @@
+(function ($) {
+    var click = function(jobId, recommender){
+        if(!JOBCENTRE.capabilities.localStorage)
+            return;
+
+        var clicks = {};
+        try {
+            clicks = JSON.parse(localStorage.getItem('recommendationInfo'));
+        } catch (e) {
+            // Continue if it fails
+        }
+
+        if(clicks == null)
+            clicks = {};
+
+        clicks[jobId] = {'recommender': recommender, 'timestamp': new Date().getTime()};
+        localStorage.setItem('recommendationInfo', JSON.stringify(clicks));
```

Comment:
How about we move all of this stuff up inside the try catch? That way, even if `clicks` is not an object (i.e. it's a string or an array), this code will fail gracefully.